packages/flutter/test/material/tabs_test.dart

```diff
@@ -2580,4 +2580,81 @@ void main() {
     await tester.pumpAndSettle();
     pageController.removeListener(pageControllerListener);
   });
+
+  testWidgets('onTabChanged responds to Tab click and Scroll Notifications', (WidgetTester tester) async {
+    const List<Tab> tabs = <Tab>[
+      Tab(text: 'Zeroth'), Tab(text: 'First'), Tab(text: 'Second')
+    ];
+    final List<int> logs = <int>[];
+
+    await tester.pumpWidget(MaterialApp(
+        home: DefaultTabController(
+          length: tabs.length,
+          child: Scaffold(
+            appBar: AppBar(
+              bottom: const TabBar(tabs: tabs),
+            ),
+            body: TabBarView(
+              children: tabs.map((Tab tab) => Center(child: Text('${tab.text} Tab'))).toList(),
+              onTabChanged: (int tabNumber) {
+                logs.add(tabNumber);
+              },
+            ),
+          ),
+        )));
+
+    // Initially logs should be empty.
+    expect(logs.isEmpty, isTrue);
+
+    // Change the Tab through TabBar.
+    await tester.tap(find.text('Second'));
+    await tester.pumpAndSettle();
+
+    expect(logs.first, 2);
+    logs.removeLast();
```

Comment:
I don't think we should remove the logs. It might be better to just have the entire log saved and test as we go. Then, the next expect would be `expect(logs.last, 1);`